File: Coding/Set1/SecondLargest.js
let arr = [1,3,4,6,4]
let largest = Number.MIN_SAFE_INTEGER
let second_largest = Number.MIN_SAFE_INTEGER
for(let num of arr){
    if(num > largest){
        second_largest = largest
        largest = num
    } else if(num > second_largest && num != largest){
        second_largest = num
    }
}
if(second_largest === Number.MIN_SAFE_INTEGER)
    console.log("there is no second highest")
else
    console.log(second_largest)

File: Coding/Python/Set1/SecondLargest.py
import sys


def second_largest(arr):
    largest = -sys.maxsize - 1
    second = -sys.maxsize - 1
    for val in arr:
        if val > largest:
            second = largest
            largest = val
        elif val > second and val != largest:
            second = val
    if second == -sys.maxsize-1:
        return -1
    return second


print(second_largest([1, 3, 4, 6, 4]))
print(second_largest([1, 1]))

